src/pynwb/core.py

```diff
@@ -47,6 +47,19 @@ def _error_on_new_warn_on_construct(self, error_msg: str):
             raise ValueError(error_msg)
         warn(error_msg)
 
+    def _error_on_new_pass_on_construct(self, error_msg: str):
```

Comment:
I added this `error_on_new_pass_on_construct` function we had discussed using for deprecated classes like `Clustering` and `ClusterWaveforms`. Just as a note, there were previously DeprecationWarnings on read of these deprecated neurodata types and with this change there would now be no warnings. I think that's ok, but wanted to confirm.

I also swapped in `error_on_new_pass_on_construct` in for some cases where different fields were deprecated/replaced by similar fields with different names, but I left `error_on_new_warn_on_construct` for checks for invalid data types. Let me know if any of those should be switched to warnings or not.

Reply:
I think that's OK too. Thanks!